docs/rules/naming-convention.md

```diff
@@ -14,151 +14,120 @@ Require names to follow specified conventions.
+# eslint @graphql-eslint/naming-convention: ['error', { types: 'PascalCase', fields: 'camelCase' }]
 
-type someTypeName {
-  f: String!
+type user {
+  first_name: String!
 }
 ```
 
 ### Correct
 
 ```graphql
-# eslint @graphql-eslint/naming-convention: ['error', { FieldDefinition: 'camelCase', ObjectTypeDefinition: 'PascalCase' }]
+# eslint @graphql-eslint/naming-convention: ['error', { types: 'PascalCase', fields: 'camelCase' }]
 
-type SomeTypeName {
-  someFieldName: String
+type User {
+  firstName: String
 }
 ```
 
 ## Config Schema
 
 The schema defines the following properties:
 
-### `FieldDefinition`
+### `types`
 
-The object must be one of the following types:
-
-* `asString`
-* `asObject`
+Includes:
 
-### `InputObjectTypeDefinition`
+- `ObjectTypeDefinition`
+- `InterfaceTypeDefinition`
+- `EnumTypeDefinition`
+- `ScalarTypeDefinition`
+- `InputObjectTypeDefinition`
+- `UnionTypeDefinition`
 
 The object must be one of the following types:
 
 * `asString`
 * `asObject`
 
-### `EnumValueDefinition`
+### `fields`
 
-The object must be one of the following types:
-
-* `asString`
-* `asObject`
-
-### `InputValueDefinition`
-
-The object must be one of the following types:
-
-* `asString`
-* `asObject`
+Includes:
 
-### `ObjectTypeDefinition`
+- `FieldDefinition`
+- `InputValueDefinition`
+- `VariableDefinition`
+- `Argument`
+- `DirectiveDefinition`
 
 The object must be one of the following types:
 
 * `asString`
 * `asObject`
 
-### `InterfaceTypeDefinition`
+### `allowLeadingUnderscore` (boolean)
 
-The object must be one of the following types:
+Default: `false`
 
-* `asString`
-* `asObject`
+### `allowTrailingUnderscore` (boolean)
 
-### `EnumTypeDefinition`
+Default: `false`
 
-The object must be one of the following types:
+### `overrides` (object)
 
-* `asString`
-* `asObject`
+May contain the following `ASTNode` names:
 
-### `UnionTypeDefinition`
+- `Argument`
+- `DirectiveDefinition`
+- `EnumTypeDefinition`
+- `EnumValueDefinition`
+- `FieldDefinition`
+- `FragmentDefinition`
+- `InputObjectTypeDefinition`
+- `InputValueDefinition`
+- `InterfaceTypeDefinition`
+- `ObjectTypeDefinition`
+- `OperationDefinition`
+- `ScalarTypeDefinition`
+- `UnionTypeDefinition`
+- `VariableDefinition`
 
-The object must be one of the following types:
+> It's also possible to use a [`selector`](https://eslint.org/docs/developer-guide/selectors) that starts with `ASTNode` name
+>
+> Example: pattern property `FieldDefinition[parent.name.value=Query]` will match only fields for type `Query`
 
-* `asString`
-* `asObject`
+Pattern properties of the `overrides` object:
 
-### `ScalarTypeDefinition`
+#### `^(Argument|DirectiveDefinition|EnumTypeDefinition|EnumValueDefinition|FieldDefinition|FragmentDefinition|InputObjectTypeDefinition|InputValueDefinition|InterfaceTypeDefinition|ObjectTypeDefinition|OperationDefinition|ScalarTypeDefinition|UnionTypeDefinition|VariableDefinition)(.+)?$`
```

Comment:
It isn't apparent what this is meant to convey in the docs.

Reply:
Changed, what do think if now it's more clear?

Reply:
Ahh. Yes. I understand what you were doing now. You were communicating in regexp? 😄 

It does seem a shame to need to have 10 sections all with just a couple words changes, but not sure there's any alternative 🤷‍♂️

Reply:
Yes this was a regex, we use a package that generate markdown documentation based on json schema 🙂